core/src/main/java/io/grpc/LoadBalancer.java

```diff
@@ -66,14 +66,16 @@
   public void shutdown() { }
 
   /**
-   * Handles newly resolved addresses and service config from name resolution system.
+   * Handles newly resolved addresses and service config from name resolution system.  Sublists
+   * should be considered equivalent with an {@link EquivalentAddressGroup}, but may be flattened
+   * into a single list if needed.
    *
    * <p>Implementations should not modify the given {@code servers}.
    *
-   * @param servers the resolved server addresses. Never empty.
+   * @param servers the resolved server addresses, never empty.
    * @param config extra configuration data from naming system.
    */
-  public void handleResolvedAddresses(List<ResolvedServerInfo> servers, Attributes config) { }
+  public void handleResolvedAddresses(List<List<ResolvedServerInfo>> servers, Attributes config) { }
```

Comment:
The generic thing above pointed out that this should probably become `List<? extends List<ResolvedServerInfo>>`, because that would allow `List<ArrayList<ResolvedServerInfo>>` to be passed in. Ditto in NameResolver.